src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -1030,21 +1040,38 @@ private static CachedCodeEntry LookupCachedAndUpdate(CachedCodeEntryKey key)
         {
             lock (s_livecode)
             {
-                for (LinkedListNode<CachedCodeEntry> current = s_livecode.First; current != null; current = current.Next)
+                s_livecode.TryGetValue(key, out var entry);
```

Comment:
In a common case (possibly the most common?), the regex will be the last used regex. Would it make sense to test whether `s_livecode_first == entry` (and return immediately if true) before looking up in the dictionary, etc.?

Reply:
if you do this then CachePromoteLinkedListEntry would be repeating that test, so you would probably move the check out into the other caller also (CacheCode ) even though in that code path it's unlikely to be true - since we aren't asked to cache anything if it was found in the cache.

Reply:
Yes, reasonable, in case of the same regex made many times. Adding this